libraries/AP_HAL/Util.cpp

```diff
@@ -46,3 +47,84 @@ int AP_HAL::Util::vsnprintf(char* str, size_t size, const char *format, va_list
     buf.write(0);
     return ret;
 }
+
+uint64_t AP_HAL::Util::get_system_clock_ms() const
+{
+    struct timespec ts;
+    clock_gettime(CLOCK_REALTIME, &ts);
+    return ((long long)(ts.tv_sec * 1000 + ts.tv_nsec/1000000));
+}
+
+void AP_HAL::Util::get_system_clock_utc(int32_t &hour, int32_t &min, int32_t &sec, int32_t &ms) const
```

Comment:
We are not taking these parameters as unsigned because we are using -1 as don't care value. Although i have reduced the size to 16 bit.